Core/LinkCleaner.swift

```diff
@@ -88,4 +88,42 @@ public class LinkCleaner {
         
         return nil
     }
+    
+    public func cleanTrackingParameters(initiator: URL?, url: URL?,
+                                        config: PrivacyConfiguration = PrivacyConfigurationManager.shared.privacyConfig) -> URL? {
+        guard let url = url, !isURLExcluded(url: url, config: config, feature: .trackingParameters) else { return nil }
+        if let initiator = initiator, isURLExcluded(url: initiator, config: config, feature: .trackingParameters) {
+            return nil
+        }
+        
+        guard let urlsComps = URLComponents(url: url, resolvingAgainstBaseURL: false) else {
```

Comment:
You could make this a `var`, then remove creation of new URLComponents instance and just assign `urlsComps.queryItems = preservedParams` at the end.